components/txn_types/src/lock.rs

```diff
@@ -95,9 +128,36 @@ impl Lock {
             b.push(MIN_COMMIT_TS_PREFIX);
             b.encode_u64(self.min_commit_ts.into_inner()).unwrap();
         }
+        if self.use_async_commit {
+            b.push(ASYNC_COMMIT_PREFIX);
+            b.encode_var_u64(self.secondaries.len() as _).unwrap();
+            for k in &self.secondaries {
+                b.encode_compact_bytes(k).unwrap();
+            }
+        }
         b
     }
 
+    fn pre_allocate_size(&self) -> usize {
+        let mut size = 1
```

Comment:
It's the maximum size. Should we calculate it accurately? BTW, it's ugly... 😭

Reply:
😢Sorry  i was trying to keep minimal change to the old code. I've made it accurate. How do you think now?

Reply:
it's inevitable to be ugly... 😂